### ISSUE TITLE
Add custom Prepare time for Klipper

### ISSUE THREAD

**Issue by O-to-the-o** [SoftFever/OrcaSlicer#5796]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Total time estimation of by the time my printer takes to prepare 

### Which printers will be beneficial to this feature?

Klipper

### Describe the solution you'd like

In the Total estimation there is the Prepare Time, it seems to be dependend on the starting g-code. In klipper many moved that to the config on the klipper host. So the Start-Gcode is just Print_Start + Parameters and the estimation gives always 1s.

I'd like to get a Parameter in the Printer Settings that just sets the Prepare time to a custom value and adjust the Total Time accordingly. This would improve the estimate for me and others

### Additional context

![grafik](https://github.com/SoftFever/OrcaSlicer/assets/58134384/6e0f3317-8c78-45e0-bbbd-8513c7de9445)

**Comment by damienvdw** [comment]
Honestly, this could be good for all printers !

I'm running my BBL P1S with custom G-code and the inaccurate prep time is bothering me so much...

A "Prepare time override" setting in the printer profile would be amazing.

**Comment by HugoCLSC** [comment]
Yes! the possibility for adjusting this paremeter, in my opinion, whould further increase the accuraty of the print time estimate.

As a bonus, QoL update, because we'll be able to just see how much time it takes for the printer to actually start printing.

Adjusting this parameter, whould in fact be amazing. For every printer that uses Orca

**Comment by freiser77** [comment]
Yes, it would be beneficial. I have a printer profile for big parts where the printer idles for 10 minutes to heat the bed evenly. I do that with a G4 command, but it is not considered in the preparation time. This would allow me to circumvent the time estimation problem.

**Comment by Dnic94** [comment]
I also stumbled over this issue. Would also be interested in an enhancement.

**Comment by MistressPlague** [comment]
I would also like this.

**Comment by Minecreator200** [comment]
It would surely be a great enhancement. Especially when included in the presets.

**Comment by chimcen** [comment]
This would be great - any news here?

**Comment by BanzoV** [comment]
This would be great!

**Comment by tealblu** [comment]
Yeah, I'd appreciate this as well!

**Comment by nikosters** [comment]
This feature would be great! I can't rely on the estimate from the slicer because it takes my printer up to 20 minutes to heat up and calibrate the bed

**Comment by Kiiiddd** [comment]
Yeah just a field in the Machine->Basic Tab where you could insert a Field for how long your printer takes to prepare would be nice. Wouldn't be super accurate with some cases where you have to preheat

**Comment by Ken6432** [comment]
Perhaps the setting should be placed in the filament settings.  It seems like the prepare time is driven mostly by the time it takes for the bed and chamber to heat up which is more filament driven than machine driven.

**Comment by broncosis** [comment]
I'll also say that this would be a nice Quality of life improvement even if its just a number we can put in ourselves doesn't need to be crazy complicated

**Comment by AndyHazz** [comment]
Or, as it's designed to look at the start g-code and work out prepare time from that, allow a special comment in that existing field (something obvious like `; Prepare time = 300`) to allow the prepare time to be overidden ...

**Comment by tastyratz** [comment]
I also was really surprised about this not existing as a printer setting for "additional prepare time"

We have filament unload, load, and toolchange time. Why not 1 more?

**Comment by mcollins1290** [comment]
I want this as well! I know at least 4 or 5 of my friends who have asked me whether I knew whether there was a way to manually set this, this would be such a great feature!!!!

**Comment by danad21** [comment]
I've stumbled upon this last year and now its bothering me again!

I suggest a field where we can input recorded time (for easy setup) or the whole gcode/bed-leveling duration/degrees per second heat up of extruder and bed/outer bed.

**Comment by GalacticBossanova** [comment]
Just adding my $.02, as I would also love this. If it can't read the actual gcode to give an accurate estimate of prep time, a manual-entry field where we can say, "Eh, I know I always run bed leveling and heating from scratch takes X amount of time, so I'll just put 5:00 in there" would be nice.

**Comment by iinsaane** [comment]
I would also like this. Based on gcode, orca guesses 26s, but total is ~6:30 with the sv08 quad gantry calibration and bed mesh

**Comment by dragoonmc** [comment]
Even if you don't want to add the GUI spot for setting this value can we do it like external flush?
; PREPARE_TIME 60

**Comment by SEKCobra** [comment]
I want to support this as well, as the bed levelling and heatup adds significant runtime to my prints and I would like that reflected in estimates.

**Comment by myersvx** [comment]
I too would like this as a feature! Just a spot to add prep time. Preferably to filament. but if not then to printer. even if only, as previously suggested, as a value to set in the start gcode section.

**Comment by BigFudgeDD** [comment]
+1

I would like that feature, too! The preparation time for my P2S ist 6:30 min, due to it's ridiculous calibration sequence. That's quite often longer or at least on par with the actual print time. The estimated time for the P2S is 13 s. Same picture with my RatRig running Klipper. Estimate is 1 s, the actual time is somewhere in the ballpark of 3...4 min.

Alternatively to let the user set the time manually, I would suggest to just ditch the preparation time estimate alltogether. I like the idea of calculating the time based on the gcode, but an estimate, which is THAT far off reality, simply isn't usefull at all.

Thank you
